nomad/constants.py

```diff
@@ -1,6 +1,12 @@
 # -*- coding: utf-8 -*-
 
 
+# PROJECTS
+# --
+
+METADATA_BASE_DIR = '.nomad'
```

Comment:
The goal here is to store a unique ID associated with each container managed by nomad.

Currently we are using a custom identifier that is associated with our container using the LXD's metadata system (see [container.py#L241](https://github.com/lxd-nomad/lxd-nomad/blob/master/nomad/container.py#L241)). The name of each container is generated at creation time by checking if there clashes with existing containers, because we want our container names to be unique (see [container.py#L205](https://github.com/lxd-nomad/lxd-nomad/blob/master/nomad/container.py#L205)). The problem with this approach is that it forces us to things like this to retrieve a container that is already created (see [container.py#L193](https://github.com/lxd-nomad/lxd-nomad/blob/master/nomad/container.py#L193)): 

```python
container = None
for _container in self.client.containers.all():
    if _container.config.get('user.nomad.nid') == str(self.nid):
        container = _container
```

This is not the good approach because we can't really get a specific container managed by nomad without checking the metadata of all the available containers. We should use the `get` method provided by pyLXD in order to retrieve a container by its name (see https://github.com/lxc/pylxd/blob/master/doc/source/containers.rst#examples), which I think is the preferred way of doing this. But we can't do this currently because we don't know what is the name of the container we are working on (at the LXD level). We need to know what are the names of our containers in order to properly retrieve them.

It is true that we can use the LXD's metadata system to store config values associated with each containers. But we need to store the identifier that allows us to get our container by its name outside this system (otherwise we cannot retrieve our container directly with the `get` method mentioned proviously). This is why this "ID" has to be stored outside the LXD's metadata system, for example in a hidden metadata file at the root of the project's folder.